misc/multiaddr/src/from_url.rs

```diff
@@ -0,0 +1,145 @@
+use crate::{Multiaddr, Protocol};
+use std::{error, fmt, iter, net::IpAddr};
+
+/// Attempts to parse an URL into a multiaddress.
+///
+/// # Example
+///
+/// ```
+/// let addr = parity_multiaddr::from_url("ws://127.0.0.1:8080/").unwrap();
+/// assert_eq!(addr, "/ip4/127.0.0.1/tcp/8080/ws".parse().unwrap());
+/// ```
+///
+pub fn from_url(url: &str) -> std::result::Result<Multiaddr, FromUrlErr> {
```

Comment:
Assuming we would use the `url` crate, how about a `TryFrom` impl for its `Url` type?